### PR TITLE
doc: copy editing in the hld topics

### PR DESCRIPTION
**Pull request by fitchbe**
Copy editing in several HLD topics. (More to follow in subsequent PRs.)

Signed-off-by: Benjamin Fitch <benjamin.fitch@intel.com>

### CODE REVIEW

File: doc/developer-guides/hld/atkbdc-virt-hld.rst [hunk 1]
```diff
@@ -49,7 +49,7 @@ PS2 Mouse Emulation
 *******************
 
 ACRN supports AT keyboard controller for PS2 mouse that can be accessed through I/O ports(0x60 and 0x64).
```
Comment (gvancuts) [inline doc/developer-guides/hld/atkbdc-virt-hld.rst:51]:
```suggestion
ACRN supports AT keyboard controller for PS2 mouse that can be accessed through I/O ports (0x60 and 0x64).
```

File: doc/developer-guides/hld/hld-APL_GVT-g.rst [hunk 2]
```diff
@@ -116,9 +116,9 @@ API Forwarding
 
 API forwarding, or a split driver model, is another widely-used I/O
 virtualization technology. It has been used in commercial virtualization
-productions, for example, VMware*, PCoIP*, and Microsoft* RemoteFx*.
+productions such as VMware*, PCoIP*, and Microsoft* RemoteFx*.
 It is a natural path when researchers study a new type of
-I/O virtualization usage, for example, when GPGPU computing in VM was
+I/O virtualization usage—for example, when GPGPU computing in a VM was
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-APL_GVT-g.rst:121]: Is this a hyphen in between `usage` and `for`?

File: doc/developer-guides/hld/hld-APL_GVT-g.rst [hunk 3]
```diff
@@ -131,7 +131,7 @@ The architecture of API forwarding is shown in :numref:`api-forwarding`:
    API Forwarding
 
 A frontend driver is employed to forward high-level API calls (OpenGL,
-Directx, and so on) inside a VM, to a Backend driver in the Hypervisor
+Directx, and so on) inside a VM to a Backend driver in the Hypervisor
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-APL_GVT-g.rst:134]:
```suggestion
DirectX, and so on) inside a VM to a Backend driver in the Hypervisor
```

Comment (gvancuts) [inline doc/developer-guides/hld/hld-APL_GVT-g.rst:134]: "Backend" is spelled with a capital "B" while "frontend" (earlier) does not use a capital "f". Should we unify this?

File: doc/developer-guides/hld/hld-APL_GVT-g.rst [hunk 4]
```diff
@@ -311,18 +311,18 @@ load and run-time phases ranging in tens of thousands per second.
...
 ACRN hypervisor, with Service VM as the privileged VM, and multiple user
-guests. A GVT-g device model working with the ACRN hypervisor,
+guests. A GVT-g device model working with the ACRN hypervisor
 implements the policies of trap and passthrough. Each guest runs the
 native graphics driver and can directly access performance-critical
 resources: the Frame Buffer and Command Buffer, with resource
-partitioning (as presented later). To protect privileged resources, that
-is, the I/O registers and PTEs, corresponding accesses from the graphics
-driver in user VMs are trapped and forwarded to the GVT device model in
+partitioning (as presented later). To protect privileged resources—that
```
Comment (gvancuts) [inline doc/developer-guides/hld/hld-APL_GVT-g.rst:323]: We do not need spaces around the hyphen?

File: doc/developer-guides/hld/ahci-hld.rst [hunk 5]
```diff
@@ -4,37 +4,37 @@ AHCI Virtualization in Device Model
...
+DM through the PCI BAR, and HBA can inject MSI interrupts through the PCI
 framework.
 
 When the application in the User VM reads data from /dev/sda, the request will
 send through the AHCI driver and then the PCI driver. The User VM will trap to
-hypervisor, and hypervisor dispatch the request to DM. According to the
-offset in the BAR, the request will dispatch to port control handler.
-Then the request is parse to a block I/O request which can be processed
-by Block backend model.
+the hypervisor, and the hypervisor will dispatch the request to the DM.
```
[the remark this answers is not in the document]
Comment (fitchbe) [reply doc/developer-guides/hld/ahci-hld.rst:26]: Agreed; wasn't sure whether this was "hypervisor lingo". Will fix.

File: doc/developer-guides/hld/ahci-hld.rst [hunk 6]
```diff
@@ -4,37 +4,37 @@ AHCI Virtualization in Device Model
...
 HBA is registered to the PCI system with device id 0x2821 and vendor id
-0x8086. Its memory registers are mapped in BAR 5. It only supports 6
-ports (refer to ICH8 AHCI). AHCI driver in the User VM can access HBA in DM
-through the PCI BAR. And HBA can inject MSI interrupts through the PCI
+0x8086. Its memory registers are mapped in BAR 5. It supports only six
+ports (refer to ICH8 AHCI). The AHCI driver in the User VM can access HBA in
+DM through the PCI BAR, and HBA can inject MSI interrupts through the PCI
 framework.
 
 When the application in the User VM reads data from /dev/sda, the request will
```
[the remark this answers is not in the document]
Comment (fitchbe) [reply doc/developer-guides/hld/ahci-hld.rst:24]: Yep; will fix.

File: doc/developer-guides/hld/ahci-hld.rst [hunk 7]
```diff
@@ -4,37 +4,37 @@ AHCI Virtualization in Device Model
...
 send through the AHCI driver and then the PCI driver. The User VM will trap to
-hypervisor, and hypervisor dispatch the request to DM. According to the
-offset in the BAR, the request will dispatch to port control handler.
-Then the request is parse to a block I/O request which can be processed
-by Block backend model.
+the hypervisor, and the hypervisor will dispatch the request to the DM.
+According to the offset in the BAR, the request will dispatch to the port
```
[the remark this answers is not in the document]
Comment (fitchbe) [reply doc/developer-guides/hld/ahci-hld.rst:27]: Thanks; will fix.

File: doc/developer-guides/hld/atkbdc-virt-hld.rst [hunk 8]
```diff
@@ -19,7 +19,7 @@ The PS2 port is a 6-pin mini-Din connector used for connecting keyboards and mic
 PS2 Keyboard Emulation
 **********************
 
-ACRN supports AT keyboard controller for PS2 keyboard that can be accessed through I/O ports(0x60 and 0x64). 0x60 is used to access AT keyboard controller data register, 0x64 is used to access AT keyboard controller address register.
+ACRN supports AT keyboard controller for PS2 keyboard that can be accessed through I/O ports(0x60 and 0x64). 0x60 is used to access AT keyboard controller data register; 0x64 is used to access AT keyboard controller address register.
```
[the remark this answers is not in the document]
Comment (fitchbe) [reply doc/developer-guides/hld/atkbdc-virt-hld.rst:22]: Thanks; will fix.

File: doc/developer-guides/hld/hld-APL_GVT-g.rst (hunk 3 above)
Comment (fitchbe) [reply doc/developer-guides/hld/hld-APL_GVT-g.rst:134]: Agreed; I'll make them all lowercase.

File: doc/developer-guides/hld/hld-APL_GVT-g.rst (hunk 2 above)
Comment (fitchbe) [reply doc/developer-guides/hld/hld-APL_GVT-g.rst:121]: Those are em dashes, and when rendered they'll be nice and wide (not like hyphens).

File: doc/developer-guides/hld/hld-APL_GVT-g.rst (hunk 4 above)
Comment (fitchbe) [reply doc/developer-guides/hld/hld-APL_GVT-g.rst:323]: Don't need spaces around an em dash. (Some people use the narrower en dash with spaces around it.)

File: doc/developer-guides/hld/hld-APL_GVT-g.rst (hunk 2 above)
Comment (gvancuts) [reply doc/developer-guides/hld/hld-APL_GVT-g.rst:121]: Thanks for the clarification 👍

File: doc/developer-guides/hld/hld-APL_GVT-g.rst (hunk 3 above)
Comment (fitchbe) [reply doc/developer-guides/hld/hld-APL_GVT-g.rst]: Fixed.

File: doc/developer-guides/hld/atkbdc-virt-hld.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (fitchbe) [reply doc/developer-guides/hld/atkbdc-virt-hld.rst]: Fixed.

**Review by gvancuts — APPROVED** [review]
Thanks!

**Pull request merged**